Watchman.Discord/Areas/Users/Services/WarnService.cs

```diff
@@ -0,0 +1,70 @@
+﻿using Devscord.DiscordFramework.Middlewares.Contexts;
+using Devscord.DiscordFramework.Services;
+using MongoDB.Driver;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Watchman.Cqrs;
+using Watchman.Discord.Areas.Users.BotCommands;
+using Watchman.DomainModel.Messages;
+using Watchman.DomainModel.Users;
+using Watchman.DomainModel.Users.Commands.Handlers;
+using Watchman.DomainModel.Users.Queries;
+using Watchman.DomainModel.Warnings;
+using Watchman.DomainModel.Warnings.Commands;
+using Watchman.DomainModel.Warnings.Queries;
+
+namespace Watchman.Discord.Areas.Protection.Services
+{
+    public class WarnService
+    {
+        private readonly UsersService _usersService;
+        private readonly ICommandBus _commandBus;
+        private readonly IQueryBus _queryBus;
+
+        public WarnService(UsersService usersService, ICommandBus commandBus, IQueryBus queryBus)
+        {
+            this._usersService = usersService;
+            this._commandBus = commandBus;
+            this._queryBus = queryBus;
+        }
+
+        public Task AddWarnToUser(AddWarnCommand command, Contexts contexts, UserContext targetUser)
+        {
+            var dbCommand = new AddWarnEventCommand(contexts.User.Id, targetUser.Id, command.Reason, contexts.Server.Id);
```

Comment:
```suggestion
            var command = new AddWarnEventCommand(contexts.User.Id, targetUser.Id, command.Reason, contexts.Server.Id);
```

Reply:
nie mogę tak bo jest dwuznaczność z argumentem funkcji

Reply:
no dbCommand to zła nazwa mimo wszystko

Reply:
aktualnie mamy 2 poziomy query/commandów, jeden to te od bota a drugie to te wywołujące akcje w bazie danych

ta konkretna nazwa dbCommand jest zła, czy ogólnie nazwa odwołująca się do tych 2 poziomów które wymieniłem jest zła i powinno być od czegoś innego, na przykład po prostu 'addWarnEventCommand'?

Reply:
to nie są operacje na bazie tylko na domenie
dopiero domena podejmuje decyzje z czego korzysta
ale równie dobrze to mogą być operacje na pliku albo na pamięci 

także wszelkie nawiązania do bazy danych w tej warstwie aplikacji nie mają sensu
bo w tej warstwie nie wiesz jeszcze co dalej sie będzie działo z danymi

Reply:
addWarnCommand / eventCommand?

Reply:
addWarnCommand albo addWarnEventCommand prędzej

Reply:
ok